mask/eolearn/tests/test_cloud_mask.py

```diff
@@ -143,50 +46,43 @@ def _check_shape(self, output, data):
 
-        # Check shape and type
-        self._check_shape(eop_clm.mask['CLM_INTERSSIM'], eop_clm.data['ALL_DATA'])
-        self._check_shape(eop_clm.data['CLP_S2C'], eop_clm.data['ALL_DATA'])
-        self.assertTrue(eop_clm.mask['CLM_INTERSSIM'].dtype == bool)
-        self.assertTrue(eop_clm.data['CLP_S2C'].dtype == np.float32)
-
-        # Compare mean cloud coverage with provided reference
-        mean_clm_provided = np.mean(eop_clm.mask['CLM'])
-        mean_clp_provided = np.mean(eop_clm.data['CLP'])
-        self.assertAlmostEqual(np.mean(eop_clm.mask['CLM_INTERSSIM']), mean_clm_provided, places=2)
-        self.assertAlmostEqual(np.mean(eop_clm.data['CLP_S2C']), mean_clp_provided, places=2)
+        # the masks and pseudo-probabilities should be equal
+        self.assertTrue(np.array_equal(eop_clm.mask['CLM_TEST'], eop_clm.mask['CLM_S2C']))
+        self.assertTrue(np.array_equal(eop_clm.data['CLP_TEST'], eop_clm.data['CLP_S2C']))
 
+    def test_cloud_coverage_downscaled(self):
         # Classifier is run on downscaled version of data array
-        add_tcm = AddMultiCloudMaskTask(data_feature='ALL_DATA',
-                                        processing_resolution=120,
-                                        mono_features=('CLP_S2C', None),
-                                        mask_feature='CLM_INTERSSIM',
-                                        average_over=16,
-                                        dilation_size=8)
+        add_tcm = CloudMaskTask(data_feature='ALL_DATA',
+                                processing_resolution=120,
+                                mono_features=('CLP_TEST', 'CLM_TEST'),
+                                mask_feature=None,
+                                average_over=8,
+                                dilation_size=4)
         eop_clm = add_tcm(self.eop)
 
         # Check shape and type
-        self._check_shape(eop_clm.mask['CLM_INTERSSIM'], eop_clm.data['ALL_DATA'])
-        self._check_shape(eop_clm.data['CLP_S2C'], eop_clm.data['ALL_DATA'])
-        self.assertTrue(eop_clm.mask['CLM_INTERSSIM'].dtype == bool)
-        self.assertTrue(eop_clm.data['CLP_S2C'].dtype == np.float32)
+        self._check_shape(eop_clm.mask['CLM_TEST'], eop_clm.data['ALL_DATA'])
+        self._check_shape(eop_clm.data['CLP_TEST'], eop_clm.data['ALL_DATA'])
+        self.assertTrue(eop_clm.mask['CLM_TEST'].dtype == bool)
+        self.assertTrue(eop_clm.data['CLP_TEST'].dtype == np.float32)
 
         # Compare mean cloud coverage with provided reference
-        mean_clm_provided = np.mean(eop_clm.mask['CLM'])
-        mean_clp_provided = np.mean(eop_clm.data['CLP'])
-        self.assertAlmostEqual(np.mean(eop_clm.mask['CLM_INTERSSIM']), mean_clm_provided, places=2)
-        self.assertAlmostEqual(np.mean(eop_clm.data['CLP_S2C']), mean_clp_provided, places=1)
+        self.assertAlmostEqual(np.mean(eop_clm.mask['CLM_TEST']), np.mean(eop_clm.mask['CLM_S2C']), places=1)
```

Comment:
Perhaps we could compare entire arrays, not just their mean values?

Reply:
I could do that if I rewrote the existing mask in the test eopatch. Should I?

Reply:
Ah, now I understand - these masks are not even the same shape because we are testing a downscaled version. But they do have the same mean values. Then everything is ok.